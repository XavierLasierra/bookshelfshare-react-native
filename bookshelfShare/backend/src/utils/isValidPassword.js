const bcrypt = require("bcrypt");

function isValidPassword(password) {
  return bcrypt.compareSync(password, this.password);
}

module.exports = { isValidPassword };
