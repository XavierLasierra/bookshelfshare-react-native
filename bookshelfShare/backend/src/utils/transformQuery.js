function transformQuery(query) {
  let transformedQuery = {};
  const entries = Object.entries(query);
  if (entries.length > 0) {
    entries.forEach(([property, filter]) => {
      if (property.includes('_')) {
        const [propertyName, symbol] = property.split('_');
        transformedQuery = { ...transformedQuery, [propertyName]: { [`$${symbol}`]: filter } };
      } else {
        transformedQuery = { ...transformedQuery, [property]: { $regex: filter, $options: 'i' } };
      }
    });
  }
  return transformedQuery;
}

module.exports = { transformQuery };
