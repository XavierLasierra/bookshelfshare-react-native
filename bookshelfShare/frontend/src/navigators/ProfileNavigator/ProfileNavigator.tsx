import React from 'react';
import { createStackNavigator, CardStyleInterpolators } from '@react-navigation/stack';

import Profile from '../../components/Profile/Profile';
import BookResults from '../../components/BookResults/BookResults';
import BookDetail from '../../components/BookDetail/BookDetail';
import UsersList from '../../components/UsersList/UsersList';
import OtherUserProfile from '../../components/OtherUserProfile/OtherUserProfile';
import AddToShelf from '../../components/AddToShelf/AddToShelf';

export default function ProfileNavigator() {
  const Stack = createStackNavigator();
  return (
    <Stack.Navigator
      initialRouteName="Profile"
      screenOptions={{
        headerShown: false,
        cardShadowEnabled: false,
        cardOverlayEnabled: false,
        cardStyleInterpolator: CardStyleInterpolators.forFadeFromCenter
      }}
    >
      <Stack.Screen
        name="Profile"
        component={Profile}
      />
      <Stack.Screen
        name="BookResults"
        component={BookResults}
      />
      <Stack.Screen
        name="BookDetail"
        component={BookDetail}
      />
      <Stack.Screen
        name="UsersList"
        component={UsersList}
      />
      <Stack.Screen
        name="OtherUserProfile"
        component={OtherUserProfile}
      />
      <Stack.Screen
        name="AddToShelf"
        component={AddToShelf}
      />
    </Stack.Navigator>
  );
}
