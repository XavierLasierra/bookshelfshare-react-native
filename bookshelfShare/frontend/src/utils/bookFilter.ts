export function bookFilter(text: string, books: any) {
  return books.reduce((acc: any, book: any) => {
    if (book.title.toLowerCase().includes(text.toLowerCase())) return [...acc, book];
    if (book.publisher.toLowerCase().includes(text.toLowerCase())) return [...acc, book];
    if (book.isbn.ISBN13.toLowerCase().includes(text.toLowerCase())) return [...acc, book];
    if (book.isbn.ISBN10.toLowerCase().includes(text.toLowerCase())) return [...acc, book];
    if (book.authors.some((author: string) => author.toLowerCase()
      .includes(text.toLowerCase()))) return [...acc, book];

    return acc;
  }, []);
}

export function bookShelfListFilter(text: string, books: any) {
  return books.reduce((acc: any, book: any) => {
    if (book.bookData.title.toLowerCase().includes(text.toLowerCase())) return [...acc, book];
    if (book.bookData.publisher.toLowerCase().includes(text.toLowerCase())) return [...acc, book];
    if (book.bookData.isbn.ISBN13.toLowerCase().includes(text.toLowerCase())) return [...acc, book];
    if (book.bookData.isbn.ISBN10.toLowerCase().includes(text.toLowerCase())) return [...acc, book];
    if (book.bookData.authors.some((author: string) => author.toLowerCase()
      .includes(text.toLowerCase()))) return [...acc, book];

    return acc;
  }, []);
}

export function bookShelfLocationFilter(filterLocation: number[], books: any) {
  return books.filter(({ customInformation: { location } }: any) => JSON.stringify(location)
  === JSON.stringify(filterLocation));
}
