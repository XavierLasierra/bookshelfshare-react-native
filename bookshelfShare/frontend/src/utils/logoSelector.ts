import BookIcon from '../assets/bookIcon.svg';
import SearchIcon from '../assets/searchIcon.svg';
import ProfileIcon from '../assets/profileIcon.svg';

export default function logoSelector(logoName: string) {
  switch (logoName) {
    case 'SearchIcon':
      return SearchIcon;
    case 'BookIcon':
      return BookIcon;
    case 'ProfileIcon':
      return ProfileIcon;
    default:
      return BookIcon;
  }
}
