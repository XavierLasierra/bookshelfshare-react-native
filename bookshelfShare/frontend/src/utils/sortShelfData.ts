export default function sortShelfData(shelf: any) {
  return shelf?.books.reduce((acc: any, book: any) => {
    const foundShelfBox = acc.find(({ location }: any) => JSON.stringify(location)
         === JSON.stringify(book?.customInformation?.location));
    if (foundShelfBox) {
      foundShelfBox.books.push(book.bookData);
    } else {
      acc.push({
        location: book?.customInformation?.location,
        books: [book.bookData]
      });
    }
    return acc;
  }, []);
}
