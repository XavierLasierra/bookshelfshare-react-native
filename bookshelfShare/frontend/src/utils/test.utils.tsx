import React from 'react';
import { render as rtlRender } from '@testing-library/react-native';
import { Provider } from 'react-redux';

import configureStore from '../redux/store';

function render(
  component: any,
  initialState: any
) {
  const store = configureStore(initialState);

  function Wrapper({ children }: any) {
    return (
      <Provider store={store}>
        {children}
      </Provider>
    );
  }
  return rtlRender(component, { wrapper: Wrapper });
}

// re-export everything
export * from '@testing-library/react-native';
// override render method
export { render };
