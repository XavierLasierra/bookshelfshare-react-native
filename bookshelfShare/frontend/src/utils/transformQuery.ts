interface Query {
    isbn?: string,
    inauthor?: string,
    intitle?: string,
    inpublisher?: string
}

export default function transformQuery(query: Query):string {
  const queryEntries = Object.entries(query);
  const url = queryEntries
    .reduce((acc, queryElement, index) => `${acc}${queryElement[0]}=${queryElement[1]}${index < queryEntries.length - 1 ? '+' : ''}`, '');
  return `${url}&key=${process.env.GOOGLE_API_KEY}`;
}
