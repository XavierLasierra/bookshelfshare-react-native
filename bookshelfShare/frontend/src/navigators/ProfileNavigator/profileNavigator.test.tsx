import React from 'react';
import {NavigationContainer} from '@react-navigation/native';
import ProfileNavigator from './ProfileNavigator';
import {render} from '../../utils/test.utils';
import loggedUserMock from '../../mocks/loggedUser.mock';
import userBooksMock from '../../mocks/userBooks.mock';

jest.useFakeTimers();

describe('Given a ProfileNavigator component', () => {
  describe('When it is rendered', () => {
    test('Then should match the snapshot', () => {
      const initialState = {
        loggedUser: {userData: loggedUserMock.user},
        userBooks: userBooksMock,
      };
      const screen = render(
        <NavigationContainer>
          <ProfileNavigator />
        </NavigationContainer>,
        initialState,
      );
      expect(screen).toMatchSnapshot();
    });
  });
});
