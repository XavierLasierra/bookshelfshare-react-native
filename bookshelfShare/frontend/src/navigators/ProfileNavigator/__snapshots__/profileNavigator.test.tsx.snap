// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Given a ProfileNavigator component When it is rendered Then should match the snapshot 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <RNCSafeAreaProvider
    onInsetsChange={[Function]}
    style={
      Array [
        Object {
          "flex": 1,
        },
        undefined,
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "backgroundColor": "rgb(242, 242, 242)",
            "flex": 1,
          },
          undefined,
        ]
      }
    >
      <View
        onLayout={[Function]}
        style={
          Object {
            "flex": 1,
          }
        }
      >
        <View
          pointerEvents="box-none"
          style={
            Object {
              "bottom": 0,
              "left": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
            }
          }
        >
          <View
            collapsable={false}
            style={
              Object {
                "opacity": 1,
              }
            }
          />
          <View
            accessibilityElementsHidden={false}
            closing={false}
            gestureVelocityImpact={0.3}
            importantForAccessibility="auto"
            onClose={[Function]}
            onGestureBegin={[Function]}
            onGestureCanceled={[Function]}
            onGestureEnd={[Function]}
            onOpen={[Function]}
            onTransition={[Function]}
            pointerEvents="box-none"
            style={
              Array [
                Object {
                  "display": "flex",
                  "overflow": undefined,
                },
                Object {
                  "bottom": 0,
                  "left": 0,
                  "position": "absolute",
                  "right": 0,
                  "top": 0,
                },
              ]
            }
            transitionSpec={
              Object {
                "close": Object {
                  "animation": "spring",
                  "config": Object {
                    "damping": 500,
                    "mass": 3,
                    "overshootClamping": true,
                    "restDisplacementThreshold": 10,
                    "restSpeedThreshold": 10,
                    "stiffness": 1000,
                  },
                },
                "open": Object {
                  "animation": "spring",
                  "config": Object {
                    "damping": 500,
                    "mass": 3,
                    "overshootClamping": true,
                    "restDisplacementThreshold": 10,
                    "restSpeedThreshold": 10,
                    "stiffness": 1000,
                  },
                },
              }
            }
          >
            <View
              collapsable={false}
              nativeID="animatedComponent"
              pointerEvents="box-none"
              style={
                Object {
                  "flex": 1,
                  "marginTop": 0,
                }
              }
            >
              <View
                enableTrackpadTwoFingerGesture={true}
                enabled={false}
                hitSlop={
                  Object {
                    "right": -700,
                  }
                }
                maxDeltaY={20}
                minOffsetX={5}
                onHandlerStateChange={[Function]}
              >
                <View
                  needsOffscreenAlphaCompositing={true}
                  style={
                    Object {
                      "flex": 1,
                      "opacity": 1,
                    }
                  }
                >
                  <View
                    style={
                      Array [
                        Object {
                          "flex": 1,
                          "overflow": "hidden",
                        },
                        Array [
                          Object {
                            "backgroundColor": "rgb(242, 242, 242)",
                          },
                          undefined,
                        ],
                      ]
                    }
                  >
                    <View
                      style={
                        Object {
                          "flex": 1,
                          "flexDirection": "column-reverse",
                        }
                      }
                    >
                      <View
                        style={
                          Object {
                            "flex": 1,
                          }
                        }
                      >
                        <RCTSafeAreaView
                          emulateUnlessSupported={true}
                          style={
                            Object {
                              "backgroundColor": "#F5F5F5",
                              "minHeight": "100%",
                            }
                          }
                        >
                          <View
                            style={
                              Object {
                                "alignItems": "flex-end",
                                "backgroundColor": "#F5F5F5",
                              }
                            }
                          >
                            <View
                              style={
                                Object {
                                  "alignItems": "center",
                                  "backgroundColor": "#D9D8C4",
                                  "borderBottomLeftRadius": 25,
                                  "justifyContent": "center",
                                  "padding": 10,
                                  "width": "100%",
                                }
                              }
                            >
                              <IconMock
                                height={45}
                                width={45}
                              />
                            </View>
                            <View
                              style={
                                Object {
                                  "height": 0,
                                }
                              }
                            >
                              <View
                                style={
                                  Object {
                                    "backgroundColor": "#D9D8C4",
                                    "height": 20,
                                    "width": 20,
                                  }
                                }
                              >
                                <View
                                  style={
                                    Object {
                                      "backgroundColor": "#F5F5F5",
                                      "borderTopRightRadius": 25,
                                      "height": 20,
                                      "width": 20,
                                    }
                                  }
                                />
                              </View>
                            </View>
                          </View>
                          <RCTScrollView>
                            <View>
                              <View
                                style={
                                  Object {
                                    "paddingHorizontal": 30,
                                  }
                                }
                              >
                                <View
                                  style={
                                    Object {
                                      "alignItems": "center",
                                      "flexDirection": "row",
                                      "paddingVertical": 10,
                                    }
                                  }
                                >
                                  <Image
                                    source={
                                      Object {
                                        "uri": "photo",
                                      }
                                    }
                                    style={
                                      Object {
                                        "borderRadius": 100,
                                        "height": 125,
                                        "marginRight": 30,
                                        "width": 125,
                                      }
                                    }
                                  />
                                  <View
                                    style={
                                      Object {
                                        "maxWidth": 200,
                                      }
                                    }
                                  >
                                    <Text
                                      style={
                                        Object {
                                          "fontSize": 20,
                                          "fontWeight": "bold",
                                          "marginBottom": 15,
                                        }
                                      }
                                    >
                                      Xavier Lasierra
                                    </Text>
                                    <Text
                                      style={
                                        Object {
                                          "fontSize": 16,
                                        }
                                      }
                                    >
                                      xlp1998@gmail.com
                                    </Text>
                                  </View>
                                </View>
                                <View
                                  style={
                                    Object {
                                      "marginTop": 20,
                                    }
                                  }
                                >
                                  <View
                                    style={
                                      Object {
                                        "flexDirection": "row",
                                        "justifyContent": "space-between",
                                      }
                                    }
                                  >
                                    <View
                                      accessible={true}
                                      collapsable={false}
                                      focusable={true}
                                      nativeID="animatedComponent"
                                      onClick={[Function]}
                                      onResponderGrant={[Function]}
                                      onResponderMove={[Function]}
                                      onResponderRelease={[Function]}
                                      onResponderTerminate={[Function]}
                                      onResponderTerminationRequest={[Function]}
                                      onStartShouldSetResponder={[Function]}
                                      style={
                                        Object {
                                          "alignItems": "center",
                                          "borderBottomWidth": 2,
                                          "borderColor": "#ECECEC",
                                          "borderTopWidth": 2,
                                          "flexDirection": "row",
                                          "height": 50,
                                          "marginBottom": 15,
                                          "opacity": 1,
                                          "paddingHorizontal": 10,
                                          "width": 160,
                                        }
                                      }
                                      testID="followersButton"
                                    >
                                      <IconMock
                                        height={25}
                                        width={25}
                                      />
                                      <Text
                                        style={
                                          Object {
                                            "fontSize": 16,
                                            "marginLeft": 15,
                                          }
                                        }
                                      >
                                        0
                                         
                                        followers
                                      </Text>
                                    </View>
                                    <View
                                      accessible={true}
                                      collapsable={false}
                                      focusable={true}
                                      nativeID="animatedComponent"
                                      onClick={[Function]}
                                      onResponderGrant={[Function]}
                                      onResponderMove={[Function]}
                                      onResponderRelease={[Function]}
                                      onResponderTerminate={[Function]}
                                      onResponderTerminationRequest={[Function]}
                                      onStartShouldSetResponder={[Function]}
                                      style={
                                        Object {
                                          "alignItems": "center",
                                          "borderBottomWidth": 2,
                                          "borderColor": "#ECECEC",
                                          "borderTopWidth": 2,
                                          "flexDirection": "row",
                                          "height": 50,
                                          "marginBottom": 15,
                                          "opacity": 1,
                                          "paddingHorizontal": 10,
                                          "width": 160,
                                        }
                                      }
                                      testID="followingButton"
                                    >
                                      <IconMock
                                        height={25}
                                        width={25}
                                      />
                                      <Text
                                        style={
                                          Object {
                                            "fontSize": 16,
                                            "marginLeft": 15,
                                          }
                                        }
                                      >
                                        0
                                         
                                        following
                                      </Text>
                                    </View>
                                  </View>
                                  <View
                                    accessible={true}
                                    collapsable={false}
                                    focusable={true}
                                    nativeID="animatedComponent"
                                    onClick={[Function]}
                                    onResponderGrant={[Function]}
                                    onResponderMove={[Function]}
                                    onResponderRelease={[Function]}
                                    onResponderTerminate={[Function]}
                                    onResponderTerminationRequest={[Function]}
                                    onStartShouldSetResponder={[Function]}
                                    style={
                                      Object {
                                        "alignItems": "center",
                                        "borderBottomWidth": 2,
                                        "borderColor": "#ECECEC",
                                        "borderTopWidth": 2,
                                        "flexDirection": "row",
                                        "height": 50,
                                        "marginBottom": 15,
                                        "opacity": 1,
                                      }
                                    }
                                    testID="readingButton"
                                  >
                                    <IconMock
                                      height={40}
                                      width={40}
                                    />
                                    <Text
                                      style={
                                        Object {
                                          "fontSize": 16,
                                          "marginLeft": 15,
                                        }
                                      }
                                    >
                                      1
                                       
                                      book
                                       
                                      currently reading
                                    </Text>
                                  </View>
                                  <View
                                    accessible={true}
                                    collapsable={false}
                                    focusable={true}
                                    nativeID="animatedComponent"
                                    onClick={[Function]}
                                    onResponderGrant={[Function]}
                                    onResponderMove={[Function]}
                                    onResponderRelease={[Function]}
                                    onResponderTerminate={[Function]}
                                    onResponderTerminationRequest={[Function]}
                                    onStartShouldSetResponder={[Function]}
                                    style={
                                      Object {
                                        "alignItems": "center",
                                        "borderBottomWidth": 2,
                                        "borderColor": "#ECECEC",
                                        "borderTopWidth": 2,
                                        "flexDirection": "row",
                                        "height": 50,
                                        "marginBottom": 15,
                                        "opacity": 1,
                                      }
                                    }
                                    testID="readButton"
                                  >
                                    <IconMock
                                      height={40}
                                      width={40}
                                    />
                                    <Text
                                      style={
                                        Object {
                                          "fontSize": 16,
                                          "marginLeft": 15,
                                        }
                                      }
                                    >
                                      1
                                       
                                      book
                                       
                                      read
                                    </Text>
                                  </View>
                                  <View
                                    accessible={true}
                                    collapsable={false}
                                    focusable={true}
                                    nativeID="animatedComponent"
                                    onClick={[Function]}
                                    onResponderGrant={[Function]}
                                    onResponderMove={[Function]}
                                    onResponderRelease={[Function]}
                                    onResponderTerminate={[Function]}
                                    onResponderTerminationRequest={[Function]}
                                    onStartShouldSetResponder={[Function]}
                                    style={
                                      Object {
                                        "alignItems": "center",
                                        "borderBottomWidth": 2,
                                        "borderColor": "#ECECEC",
                                        "borderTopWidth": 2,
                                        "flexDirection": "row",
                                        "height": 50,
                                        "marginBottom": 15,
                                        "opacity": 1,
                                      }
                                    }
                                    testID="toReadButton"
                                  >
                                    <IconMock
                                      height={40}
                                      width={40}
                                    />
                                    <Text
                                      style={
                                        Object {
                                          "fontSize": 16,
                                          "marginLeft": 15,
                                        }
                                      }
                                    >
                                      1
                                       
                                      book
                                       
                                      to read
                                    </Text>
                                  </View>
                                  <View
                                    accessible={true}
                                    collapsable={false}
                                    focusable={true}
                                    nativeID="animatedComponent"
                                    onClick={[Function]}
                                    onResponderGrant={[Function]}
                                    onResponderMove={[Function]}
                                    onResponderRelease={[Function]}
                                    onResponderTerminate={[Function]}
                                    onResponderTerminationRequest={[Function]}
                                    onStartShouldSetResponder={[Function]}
                                    style={
                                      Object {
                                        "alignItems": "center",
                                        "borderBottomWidth": 2,
                                        "borderColor": "#ECECEC",
                                        "borderTopWidth": 2,
                                        "flexDirection": "row",
                                        "height": 50,
                                        "marginBottom": 15,
                                        "opacity": 1,
                                      }
                                    }
                                    testID="wishlistButton"
                                  >
                                    <IconMock
                                      height={40}
                                      width={40}
                                    />
                                    <Text
                                      style={
                                        Object {
                                          "fontSize": 16,
                                          "marginLeft": 15,
                                        }
                                      }
                                    >
                                      1
                                       
                                      book
                                       
                                      on wishlist
                                    </Text>
                                  </View>
                                  <View
                                    accessible={true}
                                    collapsable={false}
                                    focusable={false}
                                    nativeID="animatedComponent"
                                    onClick={[Function]}
                                    onResponderGrant={[Function]}
                                    onResponderMove={[Function]}
                                    onResponderRelease={[Function]}
                                    onResponderTerminate={[Function]}
                                    onResponderTerminationRequest={[Function]}
                                    onStartShouldSetResponder={[Function]}
                                    style={
                                      Object {
                                        "alignItems": "center",
                                        "borderBottomWidth": 2,
                                        "borderColor": "#ECECEC",
                                        "borderTopWidth": 2,
                                        "flexDirection": "row",
                                        "height": 50,
                                        "marginBottom": 15,
                                        "opacity": 1,
                                      }
                                    }
                                  >
                                    <IconMock
                                      height={35}
                                      width={40}
                                    />
                                    <Text
                                      style={
                                        Object {
                                          "fontSize": 16,
                                          "marginLeft": 15,
                                        }
                                      }
                                    >
                                      0
                                       
                                      shelves
                                    </Text>
                                  </View>
                                </View>
                                <View
                                  accessible={true}
                                  collapsable={false}
                                  focusable={true}
                                  nativeID="animatedComponent"
                                  onClick={[Function]}
                                  onResponderGrant={[Function]}
                                  onResponderMove={[Function]}
                                  onResponderRelease={[Function]}
                                  onResponderTerminate={[Function]}
                                  onResponderTerminationRequest={[Function]}
                                  onStartShouldSetResponder={[Function]}
                                  style={
                                    Object {
                                      "alignItems": "center",
                                      "backgroundColor": "#000000",
                                      "borderRadius": 10,
                                      "display": "flex",
                                      "height": 70,
                                      "justifyContent": "center",
                                      "marginTop": 50,
                                      "opacity": 1,
                                      "width": "100%",
                                    }
                                  }
                                  testID="logoutButton"
                                >
                                  <Text
                                    style={
                                      Object {
                                        "color": "#FFFFFF",
                                        "fontSize": 20,
                                        "fontWeight": "500",
                                      }
                                    }
                                  >
                                    Log out
                                  </Text>
                                </View>
                              </View>
                              <View
                                style={
                                  Object {
                                    "height": 130,
                                    "width": "100%",
                                  }
                                }
                              />
                            </View>
                          </RCTScrollView>
                        </RCTSafeAreaView>
                      </View>
                    </View>
                  </View>
                </View>
              </View>
            </View>
          </View>
        </View>
      </View>
      <View
        pointerEvents="box-none"
        style={
          Object {
            "height": 44,
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
            "zIndex": 1,
          }
        }
      />
    </View>
  </RNCSafeAreaProvider>
</View>
`;
